config/rbac/role.yaml

```diff
@@ -32,6 +59,17 @@ rules:
   - patch
   - update
   - watch
+- apiGroups:
+  - cluster.x-k8s.io
+  resources:
+  - clusters
+  - clusters/status
+  - machines
+  - machines/status
+  verbs:
+  - get
+  - list
+  - watch
```

Comment:
I kept these directives because they're additive and they can be helpful if someone wants to run the kubeadm controllers in a different environment

Reply:
If someone wants to vendor Cluster API and run a manager in a different container, they'd need these RBAC permissions, which are specified under `kubeadmconfig_controller.go` to run it